editors/linux-wps-office: New port: WPS Office for Linux

yzgyyang opened the pull request:
WPS Office is a complete office suite with PDF editor.

PR:		252736
Submitted by:	Martin Filla <freebsd@sysctl.cz> (in part)
Approved by:	lwhsu (mentor)

lwhsu commented on editors/linux-wps-office/files/wpsart:
```diff
@@ -0,0 +1,2 @@
+#! /bin/sh
+exec /compat/linux/bin/sh /opt/kingsoft/wps-office/office6/wpsart"$@"
```
no space before `"$@"` ?

lwhsu commented on editors/linux-wps-office/files/wpsd:
```diff
@@ -0,0 +1,2 @@
+#! /bin/sh
+exec /compat/linux/bin/sh /opt/kingsoft/wps-office/office6/wpssd"$@"
```
no space before `"$@"` ?

yzgyyang commented:
Note for self: improve pkg-descr ;)

lwhsu commented on editors/linux-wps-office/Makefile:
```diff
@@ -0,0 +1,55 @@
+PKGNAMEPREFIX=	linux-
+DISTNAME=	${PORTNAME}-${PORTVERSION}.XA-1.x86_64
+
+MAINTAINER=	ygy@FreeBSD.org
```
Just looking at https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=252736
Doesn't original submitter want to be the maintainer?

lwhsu commented on editors/Makefile:
```diff
@@ -185,6 +185,7 @@
     SUBDIR += libreoffice-xh
     SUBDIR += libreoffice-zu
     SUBDIR += libreoffice6
+    SUBDIR += linux-wps-office
```
Sorting.

lwhsu commented on editors/linux-wps-office/Makefile:
```diff
@@ -0,0 +1,55 @@
+SHEBANG_FILES= 	usr/bin/et \
+		usr/bin/wps \
+		usr/bin/wpspdf \
+		usr/bin/wpp
```
sorting.

yzgyyang replied:
This sorting looks fine by me?

lwhsu replied:
I think usr/bin/wpp should be placed before usr/bin/wps ?

yzgyyang replied:
Oh right.............. So obvious. Need coffee. /facepalm

lwhsu commented on editors/linux-wps-office/files/patch-opt_kingsoft_wps-office_desktops_wps-office-et.desktop:
```diff
@@ -0,0 +1,11 @@
+ Comment=Use WPS Spreadsheets to analyze manage data.
+ Comment[zh_CN]=使用WPS表格分析、管理数据
+-Exec=/usr/bin/et %F
++Exec=et %F
```
Maybe `%%PREFX%%/bin/et` and replace %%PREFIX%% with ${PREFIX} in post-patch: ?

I don't think it's very necessary but I personally would prefer this.

yzgyyang replied:
I actually think that replacing `/usr/bin` with `${PREFIX}` should be enough.

lwhsu commented on editors/linux-wps-office/files/patch-opt_kingsoft_wps-office_desktops_wps-office-pdf.desktop:
```diff
@@ -0,0 +1,11 @@
+ Comment=Kingsoft Pdf Reader
+ Comment[zh_CN]=WPS PDF阅读器
+-Exec=/usr/bin/wpspdf %F
++Exec=wpspdf %F
```
`%%PREFX%%` ?

lwhsu commented on editors/linux-wps-office/files/patch-opt_kingsoft_wps-office_desktops_wps-office-prometheus.desktop:
```diff
@@ -0,0 +1,11 @@
+ Comment=Use WPS Writer to office work.
+ Comment[zh_CN]=使用 WPS 2019进行办公
+-Exec=/usr/bin/wps %F
++Exec=wps %F
```
`%%PREFX%%` ?

lwhsu commented on editors/linux-wps-office/files/patch-opt_kingsoft_wps-office_desktops_wps-office-wpp.desktop:
```diff
@@ -0,0 +1,11 @@
+ Comment=Use WPS Presentation to edit and play presentations.
+ Comment[zh_CN]=使用 WPS 演示编辑、播放演示文稿
+-Exec=/usr/bin/wpp %F
++Exec=wpp %F
```
`%%PREFX%%` ?

lwhsu commented on editors/linux-wps-office/files/patch-opt_kingsoft_wps-office_desktops_wps-office-wps.desktop:
```diff
@@ -0,0 +1,11 @@
+ Comment=Use WPS Writer to edit articles and reports.
+ Comment[zh_CN]=使用 WPS 文字编写报告，排版文章
+-Exec=/usr/bin/wps %U
++Exec=wps %U
```
`%%PREFX%%` ?

lwhsu commented on editors/linux-wps-office/files/patch-usr_bin_et:
```diff
@@ -0,0 +1,11 @@
+ 	gInstallPath=${gBinPath}
+ else
+-	gInstallPath=/opt/kingsoft/wps-office
++	gInstallPath=/compat/linux/opt/kingsoft/wps-office
```
`/compat/linux` should be `%%LINUXBASE%%` and replaced with `${LINUXBASE}` in `post-patch:` target.

Another way is just using replace to `s,gInstallPath=/opt/,gInstallPath=${LINUXBASE}/opt/,` for those files in `post-patch:` target. This might be a cleaner way.

But I still prefer if upstream can modify the code to support different install prefix.

yzgyyang replied:
Ah, will do.

> But I still prefer if upstream can modify the code to support different install prefix.

Commercial software from mainland, not really an upstream ;)

lwhsu commented on editors/linux-wps-office/files/patch-usr_bin_wpp:
```diff
@@ -0,0 +1,11 @@
+ 	gInstallPath=${gBinPath}
+ else
+-	gInstallPath=/opt/kingsoft/wps-office
++	gInstallPath=/compat/linux/opt/kingsoft/wps-office
```
`%%LINUXBASE%%`

lwhsu commented on editors/linux-wps-office/files/patch-usr_bin_wps:
```diff
@@ -0,0 +1,11 @@
+ 	gInstallPath=${gBinPath}
+ else
+-	gInstallPath=/opt/kingsoft/wps-office
++	gInstallPath=/compat/linux/opt/kingsoft/wps-office
```
`%%LINUXBASE%%`

lwhsu commented on editors/linux-wps-office/files/patch-usr_bin_wpspdf:
```diff
@@ -0,0 +1,11 @@
+ 	gInstallPath=${gBinPath}
+ else
+-	gInstallPath=/opt/kingsoft/wps-office
++	gInstallPath=/compat/linux/opt/kingsoft/wps-office
```
`%%LINUXBASE%%`

lwhsu commented on editors/linux-wps-office/Makefile:
```diff
@@ -0,0 +1,55 @@
+EXTRACT_SUFX=   .rpm
+NO_BUILD=	yes
+NO_WRKSUBDIR=	yes
+PLIST_SUB=	LINUXBASE=${LINUXBASE}
```
Maybe we can move this to Uses/linux.mk and clean up other ports :)

yzgyyang replied:
Yeah, we could potentially do this in another PR/Phabricator review. I am trying hard to think of a counterexample why we shouldn't do this, but can't.

lwhsu replied:
Yes no need to do this immediately, but is a good thing to do after adding this port. :)

yzgyyang commented:
@lwhsu This is good for another round of review/testing ;)

lwhsu commented on editors/linux-wps-office/Makefile:
```diff
@@ -0,0 +1,64 @@
+.for d in ${WPS_DESK}
+	@${REINPLACE_CMD} -e 's|/usr/bin|${PREFIX}/bin|' ${WRKSRC}/opt/kingsoft/${PORTNAME}/desktops/${PORTNAME}-${d}.desktop
+.endfor
+.for WPS_BINARY in ${WPS_BINS}
```
Just being nitpicking, `WPS_BINARY` looks a global variable to me so I am confused at the first sight, especially it is longer than `WPS_BINS`. :-)

Just a preference/style question.

lwhsu replied:
For me I'll just use names like f, file, b, binary or so in lower case.

yzgyyang replied:
Sure - this line is maintainer-submitted while I added the above one, will make them consistent.

lwhsu commented:
you might want to try DISABLE_LICENSES=yes in your local poudriere to test.

yzgyyang commented:
@lwhsu This is good for another round of review and commit upon approval. CI build for amd64 is passing. Feature tests are documented in the PR (252736), along with some discussions.

yzgyyang committed:
editors/linux-wps-office: New port: WPS Office for Linux

WPS Office is a complete office suite with PDF editor.

PR:		252736
Submitted by:	Martin Filla <freebsd@sysctl.cz> (in part)
Approved by:	lwhsu (mentor)
